sickbeard/webserve.py

```diff
@@ -5006,6 +5019,13 @@ def saveProviders(self, newznab_string='', torrentrss_string='', provider_order=
                 except Exception:
                     curNzbProvider.enable_daily = 0  # these exceptions are actually catching unselected checkboxes
 
+            if hasattr(curNzbProvider, 'enable_manualsearch'):
+                try:
+                    curNzbProvider.enable_manualsearch = config.checkbox_to_value(
+                        kwargs[curNzbProvider.get_id() + '_enable_manualsearch'])
+                except Exception:
```

Comment:
These calls should be changed.  First of all caching `Exception` is generally a bad idea.  Secondly if you're already using a `try: ... except:` then catch `AttributeError` instead of checking with `hasattr`